src/test.js

```diff
@@ -446,7 +446,12 @@ Test.prototype = {
 
 	pushResult: function( resultInfo ) {
 		if ( this !== config.current ) {
-			throw new Error( "Assertion occurred after test had finished." );
+			var message = resultInfo && resultInfo.message || "";
+			var testName = this && this.testName || "";
+			var error = "Assertion " + message +
+				" occurred after test " + testName + " had finished.";
```

Comment:
What's a default message look like, say something like `assert.ok(actual)`? Is that just going to be "Assertion okay occurred..." ?
I try to optimize for the typical cases, but it's hard to say what those are. Plus, I don't have much to suggest on how we could improve it - maybe that's as simple as we need to make it.

Reply:
Indeed. Every assertion should have a default message faik, and for `ok()` that is "okay".